File: Code/Backend/innercircle/src/main/java/bu/edu/met673/api/innercircle/model/UserDevice.java
package bu.edu.met673.api.innercircle.model;

import com.google.cloud.firestore.QueryDocumentSnapshot;

public class UserDevice {
  private String userDeviceId;
  private String deviceToken;
  private String model;
  private String osVersion;

  public UserDevice(String deviceToken, String model, String osVersion) {
    this.deviceToken = deviceToken;
    this.model = model;
    this.osVersion = osVersion;
  }

  public UserDevice(QueryDocumentSnapshot device) {
    this(device.getString("deviceToken"), device.getString("model"), device.getString("osVersion"));
    this.userDeviceId = device.getId();
  }

  public String getUserDeviceId() {
    return userDeviceId;
  }

  public String getDeviceToken() {
    return deviceToken;
  }

  public void setDeviceToken(String deviceToken) {
    this.deviceToken = deviceToken;
  }

  public String getModel() {
    return model;
  }

  public void setModel(String model) {
    this.model = model;
  }

  public String getOsVersion() {
    return osVersion;
  }

  public void setOsVersion(String osVersion) {
    this.osVersion = osVersion;
  }
}

File: Code/Android/app/src/main/java/edu/bu/metcs673/project/model/user/UserDevice.kt
package edu.bu.metcs673.project.model.user

import com.google.firebase.firestore.DocumentSnapshot

class UserDevice(var userDeviceId: String?, map: Map<String, Any?>?) {

    var deviceToken: String = map?.get("deviceToken") as String
    var model: String? = null
    var osVersion: String? = null

    constructor(documentSnapshot: DocumentSnapshot): this(documentSnapshot.id, documentSnapshot.data)
    constructor(deviceToken: String): this(null, mapOf("deviceToken" to deviceToken))

    init {
        if (userDeviceId == null) {
            userDeviceId = map?.get("userDeviceId") as String?
        }
        model = map?.get("model") as String?
        osVersion = map?.get("osVersion") as String?
    }
}